_plugins/related_posts.rb

```diff
@@ -0,0 +1,62 @@
+# https://github.com/jumanji27/related_posts-jekyll_plugin
+
+require 'jekyll/post'
+
+module Jekyll
+  module RelatedPostsByTags
+    # Used to remove #related_posts so that it can be overridden
+    def self.included(klass)
+      klass.class_eval do
+        remove_method :related_posts
+      end
+    end
+
+    # Calculate related posts.
+    # Returns [<Post>]
+    def related_posts(posts)
+      return [] unless posts.size > 1
+      highest_freq = tag_freq(posts).values.max
+      related_scores = Hash.new(0)
+
+      posts.each do |post|
+        post.tags.each do |tag|
+          if self.tags.include?(tag) && post != self
+            cat_freq = tag_freq(posts)[tag]
+            related_scores[post] += (1+highest_freq-cat_freq)
```

Comment:
Surrounding space missing for operator '-'.<br>Surrounding space missing for operator '+'.